lib/SILOptimizer/Mandatory/TFDifferentiation.cpp

```diff
@@ -2407,11 +2406,24 @@ class PrimalGenCloner final : public SILClonerWithScopes<PrimalGenCloner> {
       errorOccurred = true;
       return;
     }
-    auto calleeOriginType = calleeOriginFnRef->getFunctionType();
     // Find or register a differentiation task for this function.
-    auto *newTask = context.lookUpOrRegisterDifferentiationTask(
-        calleeOriginFnRef->getReferencedFunction(), indices,
-        /*invoker*/ {ai, synthesis.task});
+    auto *calleeOriginalFn = calleeOriginFnRef->getReferencedFunction();
+    auto *newTask =
+        context.lookUpMinimalDifferentiationTask(calleeOriginalFn, indices);
+    if (!newTask) {
+      // If the callee function does not have a primitive declaration that
```

Comment:
Nit: I'd switch clause order.
```suggestion
      // If the callee function is generic and does not have a primitive
      // declaration that covers the required indices, we cannot differentiate it
      // yet so we bail out instead of registering a new differentiation task.
```